### ISSUE TITLE
`depends-on` should behave like kustomize target selectors

### ISSUE THREAD

**Issue by kriswuollett** [GoogleContainerTools/kpt#3693]
The `depends-on` annotation accepts either a cluster-scoped resource or a specific namespaced-one:

https://github.com/GoogleContainerTools/kpt/blob/fa0a9a5dd213cbf39d508e71e8dfad5aeb4cd34a/site/reference/annotations/depends-on/README.md?plain=1#L16-L29

Since nearly all resources are namespace-scoped, it would be great if the namespace-scoped value would accept an empty string to imply the resource's namespace, e.g., `rabbitmq.com/namespaces//User/example-user`, which is similar to how [kustomize targets transformations](https://github.com/kubernetes-sigs/kustomize/blob/e3981daf0ebca762fe1a29dfd39b70418c6697ba/examples/transformerconfigs/README.md?plain=1#L174-L177
).

Use case: I'm using kustomize with kpt and I'd like to avoid patching `config.kubernetes.io/depends-on` when an overlay uses a transforms the namespace name. It just doesn't seem necessary to specify the namespace here.

**Comment by mortent** [comment]
@kriswuollett Yeah, I agree a solution where an empty namespace would be interpreted as the same namespace as the current resource namespace would be useful. The actual code for the depends-on functionality is in https://github.com/kubernetes-sigs/cli-utils, so we would have to implement it there. I haven't fully considered the implications of such a change, so it might be worth coming up with a short design proposal.
